### PR TITLE
LPS-177869 [PRM] Manager only can edit MDF in draft status

### PR DESCRIPTION
**Pull request by Manda-vit**

### CODE REVIEW

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by Manda-vit** [comment]
ci:test:stable

**Comment by Manda-vit** [comment]
ci:test:stable

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by Manda-vit** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/utils/isLiferayManager.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export function isLiferayManager(roles: Role[]) {
+	const allowedRoles = [
+		RoleTypes.CHANNEL_GENERAL_MANAGER,
+		RoleTypes.CHANNEL_ACCOUNT_MANAGER,
+		RoleTypes.CHANNEL_REGIONAL_MARKETING_MANAGER,
+		RoleTypes.CHANNEL_GLOBAL_MARKETING_MANAGER,
+		RoleTypes.CHANNEL_FINANCE_MANAGER,
+	];
+
+	return roles?.some((role) => allowedRoles.includes(role.name as RoleTypes));
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/utils/isLiferayManager.ts:31]:
```suggestion
	return roles.some((role) => allowedRoles.includes(role.name as RoleTypes));
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/utils/submitForm.ts [hunk 2]
```diff
@@ -28,21 +28,47 @@ import getTotalMDFRequest from '../../../common/utils/getTotalMDFRequest';
...
 	formikHelpers: Omit<FormikHelpers<MDFRequest>, 'setFieldValue'>,
 	siteURL: string,
-	currentRequestStatus?: LiferayPicklist
+	currentRequestStatus?: LiferayPicklist,
+	accountRoles?: AccountRole[]
 ) {
 	formikHelpers.setSubmitting(true);
 
-	if (currentRequestStatus) {
+	if (currentRequestStatus && !values.id) {
```
[the remark this answers is not in the document]
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/utils/submitForm.ts:42]: @Manda-vit, check if we can remove this.

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by thaynam** [comment]
PR resent [3552](https://github.com/pt-liferay-solutions/liferay-portal/pull/3552)

**Comment by thaynam** [comment]
Waiting for [3552](https://github.com/pt-liferay-solutions/liferay-portal/pull/3552)

**Comment by Manda-vit** [comment]
ci:test:sf

**Comment by thaynam** [comment]
resent [3559](https://github.com/pt-liferay-solutions/liferay-portal/pull/3559)

**Pull request closed without merging**